controlplane/kubeadm/internal/controllers/status.go

```diff
@@ -423,6 +424,139 @@ func setRemediatingCondition(ctx context.Context, kcp *controlplanev1.KubeadmCon
 	})
 }
 
+func setAvailableCondition(_ context.Context, kcp *controlplanev1.KubeadmControlPlane, etcdIsManaged bool, etcdMembers []*etcd.Member, etcdMembersAgreeOnMemberList bool, etcdMembersAgreeOnClusterID bool, etcdMembersAndMachinesAreMatching bool, machines collections.Machines) {
+	if !kcp.Status.Initialized {
+		v1beta2conditions.Set(kcp, metav1.Condition{
+			Type:    controlplanev1.KubeadmControlPlaneAvailableV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  controlplanev1.KubeadmControlPlaneNotAvailableV1Beta2Reason,
+			Message: "Control plane not yet initialized",
+		})
+		return
+	}
+
+	if etcdIsManaged && etcdMembers == nil {
```

Comment:
nit: Maybe make this one big if branch with `if etcdIsManaged` (for this and the next 3 ifs)